MonoGame.Framework/Graphics/Vertices/VertexBuffer.DirectX.cs

```diff
@@ -138,18 +138,52 @@ void GenerateIfRequired()
                 var startBytes = startIndex * elementSizeInBytes;
                 var dataPtr = (IntPtr)(dataHandle.AddrOfPinnedObject().ToInt64() + startBytes);
 
-                var box = new SharpDX.DataBox(dataPtr, 1, 0);
+                var d3dContext = GraphicsDevice._d3dContext;
+                lock (d3dContext)
```

Comment:
I was trying to simplify it by just having one lock, but then I messed up by copying-and-pasting several more locks into the "else" block. I'll fix it by going back to more atomic locking.